librz/core/cmp.c

```diff
@@ -0,0 +1,465 @@
+// SPDX-FileCopyrightText: 2021 Dhruv Maroo <dhruvmaru007@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_cmp.h>
+#include <rz_asm.h>
+#include <rz_list.h>
+
+/**
+ * \brief Compare memory at \p addr1 with the memory at \p addr2
+ *
+ * \param core Current RzCore instance
+ * \param addr1 address to read data from
+ * \param addr2 address to read data from
+ * \param len Number of bytes to compare
+ * \return RzCompareData* A pointer to RzCompareData comparison
+ */
+RZ_API RZ_OWN RzCompareData *rz_core_cmp_mem_mem(RzCore *core, ut64 addr1, ut64 addr2, ut32 len) {
+	rz_return_val_if_fail(core, NULL);
+
+	ut8 *buf1 = malloc(len * sizeof(ut8));
+	ut8 *buf2 = malloc(len * sizeof(ut8));
+	if (!buf1 || !buf2) {
+		goto error_goto;
+	}
+	if (rz_io_nread_at(core->io, addr1, buf1, len) == -1 || rz_io_nread_at(core->io, addr2, buf2, len) == -1) {
```

Comment:
I think you should get the results of both `rz_io_nread_at` calls and compare them to also determine if the comparison is true or false. It could happen that you just partially reads from `addr1` (or `addr2`) but you don't notice.